migrate/list-convert.go

```diff
@@ -0,0 +1,181 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"os"
+	"path/filepath"
+)
+
+const (
+	MaxThreadCount = 200
+)
+
+func getKubeConfig() (*rest.Config, error) {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return nil, err
+	}
+	kubeConfig := filepath.Join(home, ".kube", "config")
+	config, err := clientcmd.BuildConfigFromFlags("", kubeConfig)
+	if err != nil {
+		if config, err = rest.InClusterConfig(); err != nil {
+			return nil, err
+		}
+	}
+	config.QPS = float32(MaxThreadCount / 2)
+	config.Burst = MaxThreadCount
+	return config, nil
+}
+
+func convertCRD(oldDriveList []directcsi.DirectCSIDrive) (newDriveList []directpv.DirectPVDrive, therror error) {
+	newDrives := []directpv.DirectPVDrive{}
+	for _, OldDrive := range oldDriveList {
+		NewDrive := directpv.DirectPVDrive{}
+
+		// TypeMeta fields
+		NewDrive.TypeMeta.Kind = "DirectPVDrive"
+		NewDrive.TypeMeta.APIVersion = "directpv.min.io/v1beta1"
+
+		// ObjectMeta fields
+		NewDrive.ObjectMeta.Name = OldDrive.ObjectMeta.Name
+		NewDrive.ObjectMeta.GenerateName = OldDrive.ObjectMeta.GenerateName
+		NewDrive.ObjectMeta.Namespace = OldDrive.ObjectMeta.Namespace
+		NewDrive.ObjectMeta.SelfLink = OldDrive.ObjectMeta.SelfLink
+		NewDrive.ObjectMeta.UID = OldDrive.ObjectMeta.UID
+		NewDrive.ObjectMeta.ResourceVersion = OldDrive.ObjectMeta.ResourceVersion
+		NewDrive.ObjectMeta.Generation = OldDrive.ObjectMeta.Generation
+		NewDrive.ObjectMeta.CreationTimestamp = OldDrive.ObjectMeta.CreationTimestamp
+		NewDrive.ObjectMeta.DeletionTimestamp = OldDrive.ObjectMeta.DeletionTimestamp
+		NewDrive.ObjectMeta.DeletionGracePeriodSeconds = OldDrive.ObjectMeta.DeletionGracePeriodSeconds
+		NewDrive.ObjectMeta.Labels = OldDrive.ObjectMeta.Labels
+		NewDrive.ObjectMeta.Annotations = OldDrive.ObjectMeta.Annotations
+		NewDrive.ObjectMeta.OwnerReferences = OldDrive.ObjectMeta.OwnerReferences
+		NewDrive.ObjectMeta.Finalizers = OldDrive.ObjectMeta.Finalizers
+		NewDrive.ObjectMeta.ZZZ_DeprecatedClusterName = OldDrive.ObjectMeta.ZZZ_DeprecatedClusterName
+		NewDrive.ObjectMeta.ManagedFields = OldDrive.ObjectMeta.ManagedFields
+
+		// Status fields
+		NewDrive.Status.Path = OldDrive.Status.Path
+		if OldDrive.Status.DriveStatus == "InUse" || OldDrive.Status.DriveStatus == "Ready" {
+			NewDrive.Status.Status = "OK"
+		}
+		NewDrive.Status.TotalCapacity = OldDrive.Status.TotalCapacity
+		NewDrive.Status.AllocatedCapacity = OldDrive.Status.AllocatedCapacity
+		NewDrive.Status.FreeCapacity = OldDrive.Status.FreeCapacity
+		NewDrive.Status.FSUUID = OldDrive.Status.FilesystemUUID
+		NewDrive.Status.NodeName = OldDrive.Status.NodeName
+		NewDrive.Status.Topology = OldDrive.Status.Topology
+		NewDrive.Status.ModelNumber = OldDrive.Status.ModelNumber
+		NewDrive.Status.Vendor = OldDrive.Status.Vendor
+		AccessTier := OldDrive.Status.AccessTier
+		if AccessTier == "Warm" {
+			NewDrive.Status.AccessTier = "Warm"
```

Comment:
we can use access tier types here

Reply:
Ok, will do, I will use the access tier types here

Reply:
Done, I am using access tier types to cast in single line all types.